api/controller.py

```diff
@@ -538,9 +543,15 @@ def authentication_for_opds_document(self):
         OPDS document in the cache, then return the cached version.
         """
         name = flask.request.library.short_name
-        if name not in self.authentication_for_opds_documents:
-            self.authentication_for_opds_documents[name] = self.auth.create_authentication_document()
-        return self.authentication_for_opds_documents[name]
+        value = self.authentication_for_opds_documents.get(name, None)
+        if value is None:
```

Comment:
This is similar to how it was before except that now it's checking if it's `None`. Why the change to this syntax?

Reply:
The old code won't work if the caching time is set to zero. I create the authentication document and stick it into `self.authentication_for_opds_documents`, where it is _immediately deleted_. So `self.authentication_for_opds_documents[name]` fails even though I just put the document into there.

Instead I make sure the document (whether retrieved from cache or created fresh) is stored in `value`. I stick `value` in the cache for next time, but it doesn't matter if it immediately gets deleted; I already have it.